README.md

```diff
@@ -1,2 +1,164 @@
+
+The contract standardizes how an HTML definition (i.e., anything inside a `<dfn>` or `h1`-`h6` element) and respective attributes and values.
+
+Specifications that conform to the contract are, with the help of reference databases, able to unambiguously cross-reference each other's definitions.
+
+## Some examples
+
+A simple example of a definition:
+
+```HTML
+<dfn id="simple-def" data-export="" data-dfn-type="dfn">
+  I'm a simple definition
+</dfn>
+```
+
+A more complex examples:
+
+```HTML
+<!-- Example of defining an abstract operation -->
+<dfn
+  id="string-parser"
+  data-export=""
+  data-lt="parse a string"
+  data-dfn-type="abstract-op">
+  string parser
+</dfn>
+
+<!-- Example of defining a WebIDL interface -->
+<dfn
+  data-export=""
+  data-dfn-type="interface"
+  id="dom-paymentrequest"
+  data-dfn-for="">
+  <code>PaymentRequest</code>
+</dfn>
+
+```
+
+## Conformance
+
+This document serves as the canonical source for how definitions are marked-up.
+
+Conforming applications to this specification are primarily authoring tools that assist in generating technical specifications. However, this specification is also here to help people who wish to mark up documents manually (or simply want to understand why various attributes mean and what they do!).
+
+Example of authoring tools (or "generators") that try to conform to this specification:
+
+- [Bikeshed](http://github.com/tabatkins/bikeshed)
+- [ReSpec](http://github.com/w3c/respec)
+- [Wattsi](https://github.com/whatwg/wattsi)
+
+Note: An accompanying test suite helps assure authoring tools conform to the contract.
+
+## Who consumes these definitions
+
+Aside from specifications internally cross-referencing their own terms and terms in other specifications, "crawlers" extract terms defined in specifications into structured data.
+
+Examples of definition crawlers include:
+
+- [Shepherd](https://dev.csswg.org/projects/shepherd)
+- [Reffy](https://github.com/w3c/reffy)
+
+Crawlers structure and categorized extracted terms into files or databases, allowing for the creation of search tools such as:
+
+- <https://respec.org/xref>
+- Bikeshed's cross-reference search.
+
+## How to get your spec indexed
+
+To have your specification indexed by a crawler, it MUST be registered with:
+
+- [browser specs](https://github.com/w3c/browser-specs/) - See [criteria for inclusion](https://github.com/w3c/browser-specs/blob/master/README.md#spec-selection-criteria).
+- [Shepherd](https://dev.csswg.org/projects/shepherd) - [Contact maintainer(s) directly](https://dev.csswg.org/users/3).
+
+## The Contract
+
+The following sections define the elements, attributes, and attribute values (along with any specific micro-syntax) that constitute the contract.
+
+### Definitions
+
+A <dfn>definition</dfn> is:
+
+- a `<dfn>` element.
+- a `h1`-`h6` element.
+
+No other HTML elements are recognized as defining a term.
+
+#### dfn
+A a dfn-definition MUST have the following attributes:
```

Comment:
I wonder if rather than `MUST` (which ideally should [apply to agent, not to models](https://www.w3.org/TR/test-methodology/#common-mistakes)) if we should declare what a conforming definition is (a dfn element with an id attribute, or or h*  element with an id and a dfn-type attribute).

Reply:
Yeah, agree. I'll change them to statements of fact.